build-system/common/utils.js

```diff
@@ -96,21 +96,24 @@ function logFiles(files) {
 }
 
 /**
- * Extracts the list of files from argv.files.
+ * Extracts the list of files from argv.files. Throws an error if no matching
+ * files were found.
  *
  * @return {Array<string>}
  */
 function getFilesFromArgv() {
-  // TODO: https://github.com/ampproject/amphtml/issues/30223
-  // Switch from globby to a lib that supports Windows.
+  if (!argv.files) {
+    return [];
+  }
+  // TODO(#30223): globby only takes posix globs. Find a Windows alternative.
   const toPosix = (str) => str.replace(/\\\\?/g, '/');
-  return argv.files
-    ? globby.sync(
-        (Array.isArray(argv.files) ? argv.files : argv.files.split(','))
-          .map((s) => s.trim())
-          .map(toPosix)
-      )
-    : [];
+  const globs = Array.isArray(argv.files) ? argv.files : argv.files.split(',');
+  const files = globby.sync(globs.map((glob) => toPosix(glob.trim())));
```

Comment:
Nit: Maybe issue multiple requests to globby, and check if any pattern doesn't match? Eg, `unit/foo/test-*.js, non-existent/test-*.js` should probably warn, too.

Reply:
Nice catch. Updated.